File: Workshops/WapMass-Workshop/Android/WapmassLearningCenter/app/src/main/java/wapmass/wapmasslearningcenter/Activities/LoginActivity.java
package wapmass.wapmasslearningcenter.Activities;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.EditText;
import android.widget.Toast;

import wapmass.wapmasslearningcenter.R;

public class LoginActivity extends AppCompatActivity {
    private Button login;
    private SharedPreferences sharedPreferences;
    private CheckBox checkBox;

    private EditText username, password;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);
        username = findViewById(R.id.username);
        password = findViewById(R.id.password);
        checkBox = findViewById(R.id.checked);
        addLoginData();

        login = findViewById(R.id.login);
        login.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String user = username.getText().toString();
                String pass = password.getText().toString();
                String data = "";
                if(checkBox.isChecked()){
                    data = "student";
                }else{
                    data = "admin";
                }
               if(login(user, pass, checkBox.isChecked())){
                   Intent intent = new Intent(LoginActivity.this, DashboardActivity.class);
                   intent.putExtra("user", data);
                   startActivity(intent);
               }else{
                   Toast.makeText(LoginActivity.this, "Login With Your Correct Credentials", Toast.LENGTH_SHORT).show();
               }
            }
        });

    }

    private void addLoginData(){
        sharedPreferences = getPreferences(Context.MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPreferences.edit();

        editor.putString("username", "admin");
        editor.putString("password", "admin");

        editor.putString("stuName", "student");
        editor.putString("stuPass", "student");

        editor.apply();

    }

    private boolean login(String username, String password, boolean isChecked){
        sharedPreferences = getPreferences(Context.MODE_PRIVATE);
        String adminName = sharedPreferences.getString("username", "N/A");
        String adminPass = sharedPreferences.getString("password", "N/A");
        String stuName = sharedPreferences.getString("stuName", "N/A");
        String stuPass = sharedPreferences.getString("stuPass", "N/A");
        return isChecked ? stuName.equals(username) && stuPass.equals(password): (adminName.equals(username) && adminPass.equals(password)) ;
    }
}

File: KotlinForAndroid/android/FarmerbusinessSchool/app/src/main/java/agroinnova/com/farmerbusinessschool/activities/LoginActivity.kt
package com.agroinnova.farmerbusinessschool.activities
import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Toast
import com.agroinnova.farmerbusinessschool.R
import com.agroinnova.farmerbusinessschool.data.DatabaseHelper
import com.agroinnova.farmerbusinessschool.models.FarmerModel
import com.agroinnova.farmerbusinessschool.models.MoneyInItemModel
import com.agroinnova.farmerbusinessschool.utils.UIutils
import kotlinx.android.synthetic.main.activity_login.*

/**
 * LoginActivity
 *  A view that presents the login screen
 *  A user can login or visit the registration page
 */
class LoginActivity : AppCompatActivity() {
   private lateinit var helper: DatabaseHelper
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login);
        helper = DatabaseHelper.getInstance(this)
//        printUser(helper.getMoneyInItems())

        UIutils.fadeIn(cardview)
        UIutils.unFocus(this);
        buttonLogin.setOnClickListener {
            val phone: String = inputEditTextPhone.text.toString().trim()
            val pincode: String = inputEditTextPinCode.text.toString().trim()
            if (
                UIutils.validateInput(phone, inputLayoutPhone, "Enter a Phone Number") &&
                UIutils.isValidPhone(phone, inputLayoutPhone) &&
                UIutils.validateInput(pincode, inputLayoutPinCode, "Enter a Pin Code") &&
                UIutils.isValidPinCode(pincode, inputLayoutPinCode)
            ) {
                if(login(phone,  pincode)){
                    startActivity(Intent(this, DashboardActivity::class.java));
                }else{
                    Toast.makeText(this, "User does not exit ", Toast.LENGTH_SHORT).show()
                }
            }
        }
        buttonRegister.setOnClickListener {
            startActivity(Intent(this, SignupActivity::class.java));
        }
    }

    /**
     *  Login  logs using his/her phone number together with  a pincode
     *  @param phone
     *  @param pincode
     */
    fun login(phone: String, pincode: String):Boolean{
        val farmerModel: FarmerModel = helper.getFarmer(phone)
        Log.i("Farmer Details ", farmerModel.phoneNumber + " "+ farmerModel.pinCode);
        return farmerModel.phoneNumber == phone && farmerModel.pinCode == pincode
    }
    fun printUser(users: List<MoneyInItemModel>) {
        for (user: MoneyInItemModel in users) {
            Log.v("DATA", " Name : " + user.name + " id : " + user.id)
        }
    }

}
